### PR TITLE
feat!: Upgrade to DCM 1.16.0

### PR DESCRIPTION
**Pull request by witwash**
#### Summary

- upgraded to DCM 1.16.0
- fixed linter issues

##### Enabled rules:
- [avoid-conditions-with-boolean-literals](https://dcm.dev/docs/rules/common/avoid-conditions-with-boolean-literals/)
- [missing-use-result-annotation](https://dcm.dev/docs/rules/common/missing-use-result-annotation/) for `Either` and `Future<Either>`

##### Rules left disabled:
- [no-magic-string](https://dcm.dev/docs/rules/common/no-magic-string/)
- [prefer-for-loop-in-children](https://dcm.dev/docs/rules/flutter/prefer-for-loop-in-children/) 

#### Testing steps

None

#### Follow-up issues

None

#### Check during review

- Verify against Jira issue.
- Is the PR over 300 additions? Consider rejecting it with advice to split it. Is it over 500 additions? It should definitely be rejected.
- Unused code removed.
- Build passing.
- Is it a bug fix? Check that it is covered by a proper test (unit or integration).

### CODE REVIEW

**Comment by ookami-kb** [comment]
- [missing-use-result-annotation](https://dcm.dev/docs/rules/common/missing-use-result-annotation/) – I was explicitly asking for this :) e.g. for `Either` and `Future<Either>`.
- [prefer-for-loop-in-children](https://dcm.dev/docs/rules/flutter/prefer-for-loop-in-children/) – looks useful, I would enable it.

**Comment by witwash** [comment]
@ookami-kb, done. I've added `Either` and `Future<Either>` to the `@useResult` annotation rule. Is there anything else that should be added as well?

**Comment by ookami-kb** [comment]
Actually, looking at our code, [prefer-for-loop-in-children](https://dcm.dev/docs/rules/flutter/prefer-for-loop-in-children/) is not very useful, in some places `map` is more readable, so let's disable it back 🤔

**Comment by witwash** [comment]
@ookami-kb, done

File: mews_pedantic/lib/analysis_options.yaml [hunk 1]
```diff
@@ -386,6 +387,10 @@ dart_code_metrics:
     # - match-positional-field-names-on-assignment
     # - member-ordering
     # - missing-test-assertion
+    - missing-use-result-annotation:
+        types:
+          - Either
+          - Future<Either>
```
Comment (ookami-kb) [inline mews_pedantic/lib/analysis_options.yaml:393]: Generics are not supported right now, let's remove it.

Comment (witwash) [reply mews_pedantic/lib/analysis_options.yaml]: @ookami-kb, done 👍

**Review by ookami-kb — APPROVED** [review]

**Review by chaeMil — APPROVED** [review]

**Pull request merged**